File: FrontEnd/src/app/models/livre.dto.ts
export interface LivreDto {
  id: number;
  titre: string;
  auteur: string;
  isbn: string;
  description: string;
  nomCategorie: string;
  noteMoyenne: number;
  nbDisponibles: number;
  commentaires: AvisDto[]; // On lie avec le petit DTO des avis
}

export interface AvisDto {
  pseudo: string;
  note: number;
  commentaire: string;
}

File: backend/src/main/java/com/example/backend/dto/LivreDto.java
package com.example.backend.dto;

import lombok.Data;

import java.math.BigDecimal;
import java.util.List;

@Data
public class LivreDto {
    private Integer id;
    private String titre;
    private String auteur;
    private String isbn;
    private String description;
    private String nomCategorie;
    private BigDecimal noteMoyenne;
    private int nbDisponibles;
    private List<AvisDto> commentaires;

    @Data
    public static class AvisDto {
        private String pseudo;
        private Integer note;
        private String commentaire;
    }
}
